pkg/streamingpromql/operators/aggregation.go

```diff
@@ -23,17 +23,38 @@ import (
 )
 
 type Aggregation struct {
-	Inner    types.InstantVectorOperator
-	Start    time.Time
-	End      time.Time
-	Interval time.Duration
-	Grouping []string
-	Pool     *pooling.LimitingPool
```

Comment:
I had made these exported previously, and I think we should keep them exported for consistency with the other operators.

My rough rule of thumb so far has been that internal implementation details (eg. `remainingGroups`) should not be exported, and the rest should be exported - the idea being that if we wanted to replace this operator with a specialised type during optimisation, everything we'd need (eg. the inner operator and the grouping options) would be available in the exported fields.

This might be a bit premature, but the other operators follow the same pattern.

Reply:
Easy enough to change later, but fair enough for consistency.